### ISSUE TITLE
ability to delete a photo added thanks to Community

### ISSUE THREAD

**Issue by plegall** [Piwigo/Piwigo-Mobile#354]
Related to #205 

It would be logic to give users the ability to delete the photos they uploaded.

**Comment by EddyLB** [comment]
Yes, but how can the app determine if a Community user can delete a photo?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by plegall** [comment]
You need to know if the current user matches the `added_by` field of the photo. Method `pwg.images.getInfo` will tell you that. Unfortunately `pwg.session.getStatus` does not tell you your `user_id` :-/ Then I also have to make sure `pwg.images.delete` is OK to let you delete a photo you uploaded, even if you're not an admin.

So, for now external apps can't fullfill the request. I have to work on Piwigo API first.

**Comment by plegall** [comment]
The upcoming version of plugin Community let the user who added the photo delete it with `pwg.images.delete`. What's missing now is the information "is this photo added by the current user?".

I wrote earlier about making it match the `added_by` returned by `pwg.images.getInfo` but we need to know the `user_id` of the current user. I can implement that 2 ways:

1. return the `user_id` in `pwg.session.getStatus`
2. in the `pwg.images.getInfo` add a return field `can_be_deleted` and `can_be_edited` (which may depend on a setting in Community)

**Comment by EddyLB** [comment]
The second way seems to be the most appropriate one.

**Comment by EddyLB** [comment]
Upon reflection, I believe that the way you implement things does not matter for the mobile apps because the app will necessarily have to call the `pwg.images.getInfo` method.

The `pwg.session.getStatus` method is always called at login and the app can easily store the `user_id`. So adopt the most appropriate (generic?) solution.

**Comment by EddyLB** [comment]
It would then be great to extend `pwg.images.getInfo` as proposed in #228 so that the app can quickly get the properties of a list of photos belonging to a user.

**Comment by EddyLB** [comment]
To determine if a Community user uploaded an image with ID `image_id`, the app must:

- store the value of the attribute `added_by` returned by the `pwg.image.uploadAsync` method as ˋuser_id` of the user account,
- retrieve the value of the attribute `added_by` returned by the `pwg.images.getInfo` method for the provided `image_id`,
- check that the image `image_id ` was `added_by` `user_id`.

This method is functionally limited. Until the user has uploaded a photo with the iOS app, all images uploaded before cannot be identified. If the cache is cleared, the images are no longer identifiable again.

We therefore need a method to retrieve the `user_id` and `pwg.session.getStatus` would be the preferable solution.

**Comment by EddyLB** [comment]
Considered fixed in v4.4 even if the retrieval of the `user_id` is not deterministic.
